src/OrleansProviders/Streams/Common/PooledCache/FixedSizeBuffer.cs

```diff
@@ -0,0 +1,85 @@
+﻿
+using System;
+
+namespace Orleans.Providers.Streams.Common
+{
+    /// <summary>
+    /// Manages a contiguous block of memory. Calls purge with itself when it is purged.
```

Comment:
This comment is a little weird. I think I understand what you mean, but you might want to rephrase a little bit. Maybe the confusion is from the method names you chose. Perhaps RequestPurge and OnPurgeRequested (or something around those lines) would make it self explanatory?